src/twisted/test/test_defer.py

```diff
@@ -1735,6 +1736,37 @@ def test_fromCoroutineRequiresCoroutine(self) -> None:
         for thing in thingsThatAreNotCoroutines:
             self.assertRaises(defer.NotACoroutineError, Deferred.fromCoroutine, thing)
 
+    def test_inlineCallbacksCancelCaptured(self) -> None:
+        """
+        Cancelling an `inlineCallbacks` correctly handles the function catching
+        the `CancelledError`.
+        """
+        canceller1Calls: List[Deferred[Any]] = []
+        canceller2Calls: List[Deferred[Any]] = []
+        d1: Deferred[Any] = Deferred(canceller1Calls.append)
+        d2: Deferred[Any] = Deferred(canceller2Calls.append)
+
+        @defer.inlineCallbacks
+        def testFunc() -> Any:
+            try:
+                yield d1
+            except Exception:
+                pass
+
+            yield d2
+
+        funcD = testFunc()
+        self.assertFalse(funcD.called)
+
+        funcD.cancel()
+
+        self.assertEqual(canceller1Calls, [d1])
```

Comment:
Maybe also add a check for canceller2Calls ?
```suggestion
        self.assertEqual(canceller1Calls, [d1])
        self.assertEqual(canceller2Calls, [])
        # Still waiting for the second deferred.
        self.assertNoResult(funcD)
```